#161342828 Configure Authors Haven to use Postgresql as DB Engine

ucheg6 opened the pull request:
#### What does this PR do?
- Configure Authors Haven to use Postgresql as DB Engine
#### Description of Task to be completed?
- Refactor app.js file to es6 javascript
- install npm packages necessary to transpile es6 codes to es5
- install Pg pg-hstore and sequelize to configure app to use postgres as database engine
- create database haven-ah with model User and migrated the table

#### How should this be manually tested?
- git clone ` https://github.com/andela/haven-ah-backend.git `
- run npm install to install necessary npm packages
- create dot-env file and setup necessary variables for database configuration
-  run ` createdb haven-ah `to create the database
- run ` sequelize:migrate ` to migrate model
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#161342828](https://www.pivotaltracker.com/story/show/161342828)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

seyi-adeleke commented on .env:
```diff
@@ -0,0 +1,5 @@
+DB_HOST=localhost
```
.env should not be in version control

seyi-adeleke commented on app.js:
```diff
@@ -1,93 +1,32 @@
 const app = express();
 
-app.use(cors());
+const port = process.env.PORT || 5000;
```
PORT number should be from the config

seyi-adeleke commented on package.json:
```diff
@@ -4,29 +4,41 @@
     "errorhandler": "^1.5.0",
-    "express": "^4.16.3",
+    "express": "^4.16.4",
     "express-jwt": "^5.3.1",
```
remove unneccsary packages plix

seyi-adeleke commented on server/config/config.js:
```diff
@@ -0,0 +1,28 @@
+    database: process.env.DB_DATABASE_DEV,
+    host: process.env.DB_HOST,
+    dialect: 'postgres',
+    
```
remove excess whitespaces

seyi-adeleke commented on server/models/User.js:
```diff
@@ -1,57 +1,66 @@
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    resetToken: {
```
The resetToken should be nullable

ucheg6 replied:
I have implemented this

seyi-adeleke commented:
Also you should be merging into develop

seyi-adeleke commented on app.js:
```diff
@@ -1,93 +1,33 @@
 const app = express();
 
-app.use(cors());
+const { port } = config.development;
```
How's this going to work on production? 🤔

seyi-adeleke commented on package.json:
```diff
@@ -4,33 +4,34 @@
   "main": "index.js",
   "scripts": {
+    "start-dev": "nodemon --exec babel-node --presets env app.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
please fix this 😢

seyi-adeleke commented on package.json:
```diff
@@ -4,33 +4,34 @@
+    "babel-core": "^6.26.3",
+    "babel-preset-env": "^1.7.0",
+    "babel-preset-stage-2": "^6.24.1",
+    "babel-register": "^6.26.0",
```
Here as well. Shouldn't your linter be catching these?

seyi-adeleke commented on server/config/config.js:
```diff
@@ -0,0 +1,28 @@
+    password: process.env.DB_PASSWORD,
+    database: process.env.DB_TEST,
+    host: process.env.DB_HOST,
+    port: process.env.DB_PORT,
```
This is no longer a DB_PORT

seyi-adeleke commented on server/models/User.js:
```diff
@@ -1,57 +1,70 @@
+    twitter: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
```
remove this excess space

Commit:
ch(Configure Authors Haven to use Postgresql as DB Engine)

- Refactor app.js file to es6 javascript
- Remove unnecessary files and codes
- install npm packages necessary to transpile es6 codes to es5
- install Pg pg-hstore and sequelize to configure app to use postgres as database engine
- create database haven-ah with model User and migrated the table
